src/website/apps/api/v2/viewsets/event.py

```
@@ -145,15 +167,53 @@ def list(self, request, *args, **kwargs):
+        response_data = serializer.data
+
+        # Cache the response for 5 minutes
+        cache.set(cache_key, response_data, 300)
+
         logger.info(
-            "EventViewSet.list: full response prepared in %.3fs", time.time() - t0)
-        return Response(serializer.data)
+            f"EventViewSet.list: full response prepared in {time.time() - t0:.3f}s")
+        return Response(response_data)
+
+    def retrieve(self, request, *args, **kwargs):
```

Ruff (0.13.1)

Unused method argument: `request`

(ARG002)